File: php/features.php
<?php
declare(strict_types=1);

// CompanyEnrich SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CompanyEnrichFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CompanyEnrichBaseFeature();
            case "test":
                return new CompanyEnrichTestFeature();
            default:
                return new CompanyEnrichBaseFeature();
        }
    }
}

File: py/features.py
# CompanyEnrich SDK feature factory

from feature.base_feature import CompanyEnrichBaseFeature
from feature.test_feature import CompanyEnrichTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CompanyEnrichBaseFeature(),
        "test": lambda: CompanyEnrichTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
